### ISSUE TITLE
Deprecations messages in symfony 5.4

### ISSUE THREAD

**Issue by Spawnrad** [aws/aws-sdk-php#2665]
### Describe the bug

Lot of deprecation messages when using this SDK with Symfony 5.4

### Expected Behavior

No deprecation message

### Current Behavior

```
User Deprecated: Method "Iterator::key()" might add "mixed" as a native return type declaration in the future. Do the same in implementation "Aws\ResultPaginator" now to avoid errors or add an explicit @return annotation to suppress this message.
User Deprecated: Method "Iterator::next()" might add "void" as a native return type declaration in the future. Do the same in implementation "Aws\ResultPaginator" now to avoid errors or add an explicit @return annotation to suppress this message.
User Deprecated: Method "Iterator::valid()" might add "bool" as a native return type declaration in the future. Do the same in implementation "Aws\ResultPaginator" now to avoid errors or add an explicit @return annotation to suppress this message.
User Deprecated: Method "Iterator::rewind()" might add "void" as a native return type declaration in the future. Do the same in implementation "Aws\ResultPaginator" now to avoid errors or add an explicit @return annotation to suppress this message.

```

### Reproduction Steps

Just run any tests with Symfony 5.4

### SDK version used

3.262.1

### Environment details (Version of PHP (`php -v`)? OS name and version, etc.)

PHP 7.4.33

**Comment by zerkms** [comment]
These warnings come from `Aws\ResultPaginator` and the trivial fix is to add typing to phpdoc for the methods (like it was done here already: https://github.com/aws/aws-sdk-php/pull/2434)

**Comment by Spawnrad** [comment]
Version 3.316.1
Issue already here

**Comment by RanVaknin** [comment]
Hi @Spawnrad @zerkms ,

We have added the `ReturnTypeWillChange` annotation in `Aws\ResultPaginator` which are meant to suppress these warnings.

@Spawnrad  I didnt quite understand your last comment, are you still experiencing this? 

Can you enable the SDK's debug logs and share them with us just in case? 
Can you re-iterate the reproduction instructions to be more clear so I can reproduce this myself?

Thanks,
Ran~

**Comment by Spawnrad** [comment]
@RanVaknin Yes i have upgrade aws sdk and have the same issue 
Here the exact message : 

> [2024-09-13T15:48:21.781344+02:00] deprecation.INFO: User Deprecated: Method "Iterator::key()" might add "mixed" as a native return type declaration in the future. Do the same in implementation "Aws\ResultPaginator" now to avoid errors or add an explicit @return annotation to suppress this message. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Method \"Iterator::key()\" might add \"mixed\" as a native return type declaration in the future. Do the same in implementation \"Aws\\ResultPaginator\" now to avoid errors or add an explicit @return annotation to suppress this message. at /var/www/backend/vendor/symfony/error-handler/DebugClassLoader.php:331)"} []
> [2024-09-13T15:48:21.787998+02:00] deprecation.INFO: User Deprecated: Method "Iterator::next()" might add "void" as a native return type declaration in the future. Do the same in implementation "Aws\ResultPaginator" now to avoid errors or add an explicit @return annotation to suppress this message. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Method \"Iterator::next()\" might add \"void\" as a native return type declaration in the future. Do the same in implementation \"Aws\\ResultPaginator\" now to avoid errors or add an explicit @return annotation to suppress this message. at /var/www/backend/vendor/symfony/error-handler/DebugClassLoader.php:331)"} []
> [2024-09-13T15:48:21.788274+02:00] deprecation.INFO: User Deprecated: Method "Iterator::valid()" might add "bool" as a native return type declaration in the future. Do the same in implementation "Aws\ResultPaginator" now to avoid errors or add an explicit @return annotation to suppress this message. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Method \"Iterator::valid()\" might add \"bool\" as a native return type declaration in the future. Do the same in implementation \"Aws\\ResultPaginator\" now to avoid errors or add an explicit @return annotation to suppress this message. at /var/www/backend/vendor/symfony/error-handler/DebugClassLoader.php:331)"} []
> [2024-09-13T15:48:21.788538+02:00] deprecation.INFO: User Deprecated: Method "Iterator::rewind()" might add "void" as a native return type declaration in the future. Do the same in implementation "Aws\ResultPaginator" now to avoid errors or add an explicit @return annotation to suppress this message. {"exception":"[object] (ErrorException(code: 0): User Deprecated: Method \"Iterator::rewind()\" might add \"void\" as a native return type declaration in the future. Do the same in implementation \"Aws\\ResultPaginator\" now to avoid errors or add an explicit @return annotation to suppress this message. at /var/www/backend/vendor/symfony/error-handler/DebugClassLoader.php:331)"} []

Juste making this call make me have the deprecated message.

        $s3Client = new S3Client([
            'region'  => getenv('AWS_REGION'),
            'version' => 'latest',
            'credentials' => [
                'key'    => getenv('AWS_KEY'),
                'secret' => getenv('AWS_SECRET'),
            ],
        ]);
        
I use symfony 5.4.

"name": "aws/aws-sdk-php",
"version": "3.321.10",

PHP 7.4.33

Thanks for your help

**Comment by RanVaknin** [comment]
Hi @Spawnrad ,

Can you update your SDK version and see if this has fixed your problem?

Thanks,
Ran~

**Comment by Spawnrad** [comment]
Hi @RanVaknin,

The issue as been solved since the update to sdk 3.324.3

Thanks a lot for your help <3
